src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -241,6 +261,32 @@ else if (!period.isEmpty()) {
         }
     }
 
+    /**
+     * Check the summary when of inline form.
+     *
+     * @param ast Javadoc root Node.
+     */
+    private void withSummaryTag(DetailNode ast) {
+        final String inlineSummaryDoc = getInlineSummarySentence(ast);
+        if (inlineSummaryDoc.isEmpty()) {
+            log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
+        }
+        else if (!period.isEmpty()) {
+            final String inlineFirstSentence = getInlineFirstSentence(ast);
+            final int endOfInlineSentence = inlineFirstSentence.lastIndexOf(period);
```

Comment:
move this two sentences to be after following IF condition, because If condition do not need evaluation of them.
we should do calculation of variables right before usage.

Reply:
Done.